### PR TITLE
fixed category field shows all categories

### PR DESCRIPTION
**Pull request by syedgalib**

### CODE REVIEW

**Comment by vairafiq** [comment]
image

No category when instant search is enabled

**Comment by vairafiq** [comment]
Screen Shot 2022-06-13 at 10 15 58 AM
Still showing all categories when Category Lazy loading is enabled

File: templates/archive/directory-type-nav.php [hunk 1]
```diff
@@ -31,7 +31,7 @@
 			<?php foreach ( $listings->listing_types as $id => $value ) : ?>
 
 				<li class="<?php echo ( ( $listings->current_listing_type === $value['term']->term_id && 'all' !== $current_directory_type ) ? 'current': '' ); ?>">
-					<a class="directorist-type-nav__link" href="<?php echo esc_url( directorist_get_directory_type_nav_url( $value['term']->slug ) ); ?>"><span class="<?php echo esc_attr( $value['data']['icon'] );?>"></span> <?php echo esc_html( $value['name'] );?></a>
+					<a class="directorist-type-nav__link" href="<?php echo esc_url( directorist_get_directory_type_nav_url( $value['term']->slug ) ); ?>" data-listing_type="<?php echo $value['term']->slug; ?>" data-listing_type_id="<?php echo $value['term']->term_id; ?>"><span class="<?php echo esc_attr( $value['data']['icon'] );?>"></span> <?php echo esc_html( $value['name'] );?></a>
```
Comment (kowsar89) [inline templates/archive/directory-type-nav.php:34]: Need proper escaping

File: includes/model/Listings.php [hunk 2]
```diff
@@ -1728,8 +1728,17 @@ public function get_wrapper_class( $class = '' ) {
 		}
 
 		public function data_atts() {
+			$atts = $this->atts;
```
Comment (kowsar89) [inline includes/model/Listings.php:1731]: what is the purpose to modify this method?

**Review by kowsar89 — CHANGES_REQUESTED** [review]

File: includes/model/Listings.php (hunk 2 above)
Comment (syedgalib) [reply includes/model/Listings.php:1731]: The Select2 controller of the category field requires directory type ID for its functionality, so extracted the directory type id from its slug and appended to the atts data.

**Pull request merged**